### PR TITLE
Improve logging / assertions system, remove fatalErrors

### PR DESCRIPTION
**Pull request by calda**
This PR updates our logging system to match Epoxy's `EpoxyLogger`, and cleaned up / centralized some of the handling for features that are not supported by the experimental rendering engine.

### CODE REVIEW

File: Sources/Public/Animation/AnimationPublic.swift [hunk 1]
```diff
@@ -69,7 +69,7 @@ extension Animation {
       return animation
     } catch {
       /// Decoding error.
-      print(error)
+      LottieLogger.shared.assertionFailure("Error when decoding animation \"\(name)\": \(error)")
```
Comment (erichoracek) [inline Sources/Public/Animation/AnimationPublic.swift:72]: Wondering if you think this should be a `warn`? I could imagine a case where you want to recover from an invalid animation and not have it crash in debug e.g. if you're writing a lottie viewer app

**Review by erichoracek — APPROVED** [review]
LGTM! Just a few small questions

File: Sources/Public/Animation/AnimationPublic.swift (hunk 1 above)
Comment (calda) [reply Sources/Public/Animation/AnimationPublic.swift:72]: Sure

Comment (calda) [reply Sources/Public/Animation/AnimationPublic.swift:72]:
```suggestion
      LottieLogger.shared.warn("Error when decoding animation \"\(name)\": \(error)")
```

**Pull request merged**